### ISSUE TITLE
Contact export is very slow with many events in the system

### ISSUE THREAD

**Issue by MartinWahnschaffeSymeda** [hzi-braunschweig/SORMAS-Project#3612]
### Problem Description
The contact detailed export has a field with the number of events where the person of the contact is involved. Gathering this information is done via a sub-query which becomes very slow (8 seconds per batch) on production systems.

### Proposed Change
It looks like we are collecting all events for the contact anyway in a second query (see getEventSummaryDetailsByContacts).

- [x] remove the event count sub query done in ContactListCriteriaBuilder (withEventInfo)

Alter the code in ContactFacadeEjb.getIndexDetailedList

- [x] Create a map contact uuid x ContactIndexDetailedDto (can also be used for the uuid list that goes into the query)
- [x] Sort the eventSummaries by event date descending
- [x] go through the eventSummaries *once* (!): 
  - [x] If the contact does not have a latest event Id yet, set it
  - [x] Increment the count of events for the contact

**Comment by MartinWahnschaffeSymeda** [comment]
Similar #3650

**Comment by StefanKock** [comment]
@MartinWahnschaffeSymeda please review the commits, but do not merge anything. I have to clarify with @MateStrysewske how to integrate it into the hotfix.

**Comment by MartinWahnschaffeSymeda** [comment]
Reviewed the code.

**Comment by StefanKock** [comment]
Merged this into hotfix 1.52.1:
```
git checkout feature-3612-avoid-eventcount-subquery
git fetch
git rebase origin/hotfix-1.52.1
git checkout hotfix-1.52.1
git merge feature-3612-avoid-eventcount-subquery
git push
git branch -D feature-3612-avoid-eventcount-subquery
git push origin --delete feature-3612-avoid-eventcount-subquery
```

**Comment by MateStrysewske** [comment]
This also needs to be done for the export.

**Comment by StefanKock** [comment]
> This also needs to be done for the export: ContactFacadeEjb:getExportList

Done and merged on hotfix 1.52.1, please review.
